src/components/Component.js

```diff
@@ -0,0 +1,13 @@
+import { $ } from "../utils/selectors.js";
+
+export default class Component {
+  constructor(app, props) {
+    this.$app = $(app);
+    this.props = props;
+  }
```

Comment:
지금 컴포넌트들이 생성된후에 공통적으로 mount, render를 호출하고 있는데, 컴포넌트 클래스에서 라이프사이클을 구현하면 이 부분도 중복을 없앨 수 있고 라이프사이클이 하위 컴포넌트에서 오용될 우려도 적어질 것 같습니다.!
```suggestion
  constructor(app, props) {
    this.$app = $(app);
    this.props = props;
    
    this.render();
    this.mount();
  }
```